onnxruntime/core/framework/session_options.h

```diff
@@ -119,7 +120,13 @@ struct SessionOptions {
   std::unordered_map<std::string, const OrtValue*> initializers_to_share_map;
 
   // See onnxruntime_c_api.h for detailed documentation.
-  Status AddInitializer(_In_z_ const char* name, _In_ const OrtValue* val) noexcept;
+  Status AddInitializer(_In_z_ const char* name, _In_ const OrtValue* val);
+
+#if !defined(ORT_MINIMAL_BUILD)  && !defined(DISABLE_EXTERNAL_INITIALIZERS)
+  // Customer supplied pre-processed data for external initializers
+  InlinedHashMap<std::string, OrtValue> external_initializers;
+  Status AddExternalInitializers(gsl::span<const std::string> names, gsl::span<const OrtValue> values);
```

Comment:
>AddExternalInitializers [](http://example.com/codeflow?start=9&length=23)

this one adds a collection of initializers and AddInitializer() adds a single one. why are they different?

Reply:
I am a little biased towards adding a collection, however, I do not have a strong opinion one way or another.